### ISSUE TITLE
Narrator fails to read the name for all the editable number boxes in "Number Box" page.

### ISSUE THREAD

**Issue by YuliKl** [microsoft/microsoft-ui-xaml#3505]
**Steps to reproduce the bug**

Pre-requisite: 
- Turn on Narrator.

1. Launch XAML Controls Gallery.
2. Activate "number box" from home page.
3. Navigate to all the editable number boxes in the page.
4. Observe narrator's description for the same.

**Expected behavior**

Narrator should read unique name, role for the all the respective number boxes.

**Screenshots**

![image](https://user-images.githubusercontent.com/13142665/97391107-ac9e8700-189b-11eb-91f9-db0966a76faf.png)

![image](https://user-images.githubusercontent.com/13142665/97391142-bd4efd00-189b-11eb-9f12-ce0045abe797.png)

**Version Info**

Xaml Controls Gallery version: 1.2.13.0

**Additional context**

Copied from [internal bug](https://microsoft.visualstudio.com/OS/_workitems/edit/25441108/)

**Comment by YuliKl** [comment]
@SavoySchuler , there was a question for you in the original issue:
What is the expected behavior here ? When Narrator lands on the NumberBox it does not read anything. However, there are two more stops, one for the header and another for the TextBox where the expression is present and they are read out. Should NumberBox have a default automation name if one is not provided ? I think providing an automation name for NumberBox will fix the issue.

**Comment by robloo** [comment]
Another duplicate of #2951 I believe.

**Comment by chingucoding** [comment]
This is not exactly the same as #2951 as textbox controls also provide a UIA name even if AutomationProperties.Nam is not set (e.g. by taking the header into account). This might be related to or a duplicate of #1853 .

**Comment by chingucoding** [comment]
I would like to look into this.
